From 7436669dde08abd5d4c6fe4716f43263b3567161 Mon Sep 17 00:00:00 2001
From: Ray Strode <rstrode@redhat.com>
Date: Mon, 16 Sep 2013 01:31:48 +0000
Subject: Revert "goaidentity: Fix deadlock in goaalarm on_cancelled"

This reverts commit f4c04ad49c29d8f619cb80ef480daa9d4ecdfa38.

It's changes which threads code runs on and is probably causing
100% cpu issues.

https://bugzilla.gnome.org/show_bug.cgi?id=705395
---
diff --git a/src/goaidentity/goaalarm.c b/src/goaidentity/goaalarm.c
index 9ccb3b6..c8992cd 100644
--- a/src/goaidentity/goaalarm.c
+++ b/src/goaidentity/goaalarm.c
@@ -256,31 +256,12 @@ goa_alarm_init (GoaAlarm *self)
   g_rec_mutex_init (&self->priv->lock);
 }
 
-static gboolean
-async_alarm_cancel_idle_cb (gpointer user_data)
-{
-  GoaAlarm *self = user_data;
-
-  clear_scheduled_wakeups (self);
-  return G_SOURCE_REMOVE;
-}
-
 static void
 on_cancelled (GCancellable *cancellable, gpointer user_data)
 {
   GoaAlarm *self = GOA_ALARM (user_data);
-  GMainContext *main_context;
-  GSource *idle_source;
-
-  main_context = g_main_context_ref_thread_default ();
 
-  idle_source = g_idle_source_new ();
-  g_source_set_priority (idle_source, G_PRIORITY_HIGH_IDLE);
-  g_source_set_callback (idle_source, async_alarm_cancel_idle_cb, g_object_ref (self), g_object_unref);
-  g_source_attach (idle_source, main_context);
-  g_source_unref (idle_source);
-
-  g_main_context_unref (main_context);
+  clear_scheduled_wakeups (self);
 }
 
 static void

